app/src/main/java/org/bspb/smartbirds/pro/backend/dto/Nomenclature.java

```diff
@@ -61,72 +56,48 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return "Nomenclature{" +
-                "type='" + type + '\'' +
-                ", label=" + label +
-                '}';
+        final StringBuffer sb = new StringBuffer("Nomenclature{");
+        sb.append("type='").append(type).append('\'');
+        sb.append(", label=").append(label);
+        sb.append(", labelId='").append(labelId).append('\'');
+        sb.append(", localeLabel='").append(localeLabel).append('\'');
+        sb.append('}');
+        return sb.toString();
     }
 
-    public static Nomenclature from(SpeciesNomenclature species) {
-        return new Nomenclature(species);
+    public static Nomenclature fromCursor(Cursor cursor, Locale locale) {
+        String data = cursor.getString(cursor.getColumnIndexOrThrow(DATA));
+        Nomenclature nomenclature;
+        if (TextUtils.isEmpty(data)) {
+            nomenclature = new Nomenclature();
+            nomenclature.type = cursor.getString(cursor.getColumnIndexOrThrow(TYPE));
+            Label label = new Label();
+            label.addValue("bg", cursor.getString(cursor.getColumnIndexOrThrow(LABEL_BG)));
+            label.addValue("en", cursor.getString(cursor.getColumnIndexOrThrow(LABEL_EN)));
+            nomenclature.label = label;
+        } else {
+            nomenclature = SBGsonParser.createParser().fromJson(data, Nomenclature.class);
+        }
+        if (nomenclature.type.startsWith("species_")) {
+            nomenclature.label = new SpeciesLabel(nomenclature.label);
+        }
+        nomenclature.localeLabel = nomenclature.label.get(locale);
+        nomenclature.labelId = cursor.getString(cursor.getColumnIndexOrThrow(LABEL_EN));
+        return nomenclature;
+    }
+
+    public static Nomenclature fromSpecies(Nomenclature species) {
+        Nomenclature nomenclature = new Nomenclature();
+        nomenclature.type = "species_" + species.type;
+        nomenclature.label = new SpeciesLabel(species.label);
+        return nomenclature;
```

Comment:
Just noticed that this doesn't fill the transient fields. Maybe it's fine, but would be easier in the future if they are filled